tests/acceptance/setup-global-a11y-hooks-test.ts

```diff
@@ -0,0 +1,109 @@
+import { module, test } from 'qunit';
+import { resolve } from 'rsvp';
+import {
+  setupGlobalA11yHooks,
+  teardownGlobalA11yHooks,
+  invokeAll,
+} from 'ember-a11y-testing/test-support';
+import { setupApplicationTest } from 'ember-qunit';
+import { visit } from '@ember/test-helpers';
+import { InvocationStrategy } from 'ember-a11y-testing/test-support/invocation-strategies';
+
+module('setupGlobalA11yHooks', function (hooks) {
+  setupApplicationTest(hooks);
+
+  module('with invokeAll', function (hooks) {
+    const TEST_INVOCATIONS_COUNT = 6;
+    const EXPECTED_AUDIT_INVOCATIONS_COUNT = 3;
+    let actualAuditInvocationsCount = 0;
+
+    function a11yAuditFake() {
+      actualAuditInvocationsCount += 1;
+      return resolve(undefined);
+    }
+
+    hooks.beforeEach(function () {
+      setupGlobalA11yHooks(invokeAll, a11yAuditFake);
+    });
+
+    hooks.afterEach(function () {
+      teardownGlobalA11yHooks();
+    });
+
+    Array(TEST_INVOCATIONS_COUNT)
```

Comment:
We should probably add a comment or two about why this is being done this way.

Reply:
What do you mean specifically? Comments don't age well, so I'd prefer to use comments sparingly.

What do you think will help here that the code can't tell you?

Reply:
Defining a test from within an iterator is an atypical pattern. I'm thinking that it would be helpful to provide some context, in that this is being done to verify invoking an nth test.

Reply:
I'm not sure it's an atypical pattern. We do this in [other places](https://github.com/ember-lifeline/ember-lifeline/blob/master/tests/unit/dom-event-listeners-test.js#L46) to test variants. In this case, we're testing the test infrastructure in order to validate that our helpers work (this addon is a test infra addon). I don't believe it warrants a comment, but could understand expanding the test name if you have suggestions.